modPrecompiled/src/org/aion/precompiled/PrecompiledSideEffects.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2017-2018 Aion foundation.
+ *
+ *     This file is part of the aion network project.
+ *
+ *     The aion network project is free software: you can redistribute it
+ *     and/or modify it under the terms of the GNU General Public License
+ *     as published by the Free Software Foundation, either version 3 of
+ *     the License, or any later version.
+ *
+ *     The aion network project is distributed in the hope that it will
+ *     be useful, but WITHOUT ANY WARRANTY; without even the implied
+ *     warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *     See the GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with the aion network project source files.
+ *     If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Contributors:
+ *     Aion foundation.
+ */
+package org.aion.precompiled;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.aion.types.Address;
+import org.aion.vm.api.interfaces.IExecutionLog;
+import org.aion.vm.api.interfaces.InternalTransactionInterface;
+import org.aion.vm.api.interfaces.TransactionSideEffects;
+
+public class PrecompiledSideEffects implements TransactionSideEffects {
+
+    private Set<Address> deleteAccounts = new HashSet<>();
+    private List<InternalTransactionInterface> internalTxs = new ArrayList<>();
+    private List<IExecutionLog> logs = new ArrayList<>();
+
+    @Override
+    public void addToDeletedAddresses(Address address) {
+        deleteAccounts.add(address);
+    }
+
+    /**
+     * Adds the collection addresses to the set of deleted accounts if addresses is non-null.
+     *
+     * @param addresses The addressed to add to the set of deleted accounts.
+     */
+    @Override
+    public void addAllToDeletedAddresses(Collection<Address> addresses) {
+        for (Address addr : addresses) {
+            if (addr != null) {
+                deleteAccounts.add(addr);
+            }
+        }
+    }
+
+    /**
+     * Adds log to the execution logs.
+     *
+     * @param log The log to add to the execution logs.
+     */
+    @Override
+    public void addLog(IExecutionLog log) {
+        logs.add(log);
+    }
+
+    /**
+     * Adds a collection of logs, logs, to the execution logs.
```

Comment:
what's the second `logs` for?

Reply:
It's meant to be read as "This function adds a collection of logs, received as the parameter `logs`, to the execution logs". This is the general pattern of the javadoc comments, copied from `SideEffects.java` in the FVM.

Reply:
(I agree that it's confusing / clumsy, though)

Reply:
I see.